com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -411,6 +411,9 @@ public static ObservationProto GetObservationProto(this ISensor sensor, Observat
                 }
             }
             observationProto.Shape.AddRange(shape);
+
+            var typeSensor = sensor as ITypedSensor;
+            observationProto.SensorType = (SensorTypeProto)typeSensor.GetSensorType();
```

Comment:
You need some sort of null check here and decide how to handle it. Either always use `Observation`, or add another `Default` enum and leave it up to the trainer to decide what to do with it (presumably treat it as an Observation anyway)